pandas/core/indexes/base.py

```diff
@@ -6989,10 +6990,10 @@ def argmax(self, axis=None, skipna: bool = True, *args, **kwargs) -> int:
 
         if not self._is_multi and self.hasnans:
             # Take advantage of cache
```

Comment:
I assume this cache comment is supposed to be tied to the `self._isnan.all()` call? Can also remove

Reply:
I had thought the cache was referring to `self.hasnans` - which is why we don't have to interrogate the values when `skipna=False`. But yea, I agree this can be removed.